tests/rejected_tx_test.go

```diff
@@ -0,0 +1,128 @@
+package tests
+
+import (
+	"context"
+	"math/big"
+	"testing"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/ethclient"
+	"github.com/stretchr/testify/require"
+)
+
+func TestRejectedTx(t *testing.T) {
+	require := require.New(t)
+
+	// start network
+	net, err := StartIntegrationTestNet(t.TempDir())
+	require.NoError(err)
+	defer net.Stop()
+
+	// create a client
+	client, err := net.GetClient()
+	require.NoError(err, "failed to get client")
+
+	chainId := getChainId(t, net)
+
+	testCases := []struct {
+		name    string
+		txMaker func(t *testing.T, account *Account, nonce uint64, price int64) *types.Transaction
+	}{
+		{name: "LegacyTx",
+			txMaker: func(t *testing.T, account *Account, nonce uint64, price int64) *types.Transaction {
+				factory := &txFactory{
+					senderKey: account.PrivateKey,
+					chainId:   chainId,
+				}
+				return factory.makeLegacyTransactionWithPrice(t, nonce, price)
+			},
+		}, {name: "AccessListTx",
+			txMaker: func(t *testing.T, account *Account, nonce uint64, price int64) *types.Transaction {
+				factory := &txFactory{
+					senderKey: account.PrivateKey,
+					chainId:   chainId,
+				}
+				return factory.makeAccessListTransactionWithPrice(t, nonce, price)
+			},
+		}, {name: "DynamicTx",
+			txMaker: func(t *testing.T, account *Account, nonce uint64, price int64) *types.Transaction {
+				factory := &txFactory{
+					senderKey: account.PrivateKey,
+					chainId:   chainId,
+				}
+				return factory.makeDynamicFeeTransactionWithPrice(t, nonce, price)
+			},
+		}, {name: "BlobTx",
+			txMaker: func(t *testing.T, account *Account, nonce uint64, price int64) *types.Transaction {
+				factory := &txFactory{
+					senderKey: account.PrivateKey,
+					chainId:   chainId,
+				}
+				return factory.makeBlobTransactionWithPrice(t, nonce, price)
+			},
+		},
+	}
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			maxFeeCap := getMaxFee(t, client)
+			newAccount := NewAccount()
+			nonce := getNonce(t, client, newAccount.Address())
+			testRejectedTx(t, net, newAccount.Address(), tc.txMaker(t, newAccount, nonce, maxFeeCap))
+		})
+	}
+}
+
+func testRejectedTx(t *testing.T, net *IntegrationTestNet, account common.Address, tx *types.Transaction) {
+	require := require.New(t)
+
+	// verify estimated cost
+	estimatedCost := tx.Gas()*tx.GasFeeCap().Uint64() + tx.Value().Uint64()
```

Comment:
To facilitate future work, estimated costs should include blobgasfee. This should be checked independently of block processing because here we are an evaluation balance being large enough for the worst case scenario. Additionally, it may turn quite difficult to find this code (tx_pool)  when implementing the blobs deep inside the block processor, doing it now will facilitate that work. 

I will update the test for blobls, make sure that they do not charge the blobgasfee